lib/chef/version_resolver.rb

```diff
@@ -101,53 +101,52 @@ def package_list
     # Finds the raw metadata info for the given platform, platform_version
     # and machine_architecture
     def find_raw_metadata_for(target_platform, target_architecture)
-      # omnitruck has some core platforms like ubuntu, windows, ...
-      # and some secondary platforms that map onto core platforms like linuxmint, scientific
-      # mapped_name will return the original name if it is not mapped
       core_platform = target_platform.mapped_name
-
-      # first make sure we have some builds available for the given platform
-      if !build_map[core_platform]
-        raise InvalidDownloadPath, "Cannot find any #{OmnitruckDist::CLIENT_NAME} versions for core platform #{core_platform}: #{friendly_error_msg}"
+      # Debug information
+      puts "Debug Info - Core Platform: #{core_platform}"
+      # puts "Debug Info - Build Map: #{build_map.inspect}"
+      # Check for the existence of builds
+      unless build_map[core_platform]
+        raise InvalidDownloadPath, "Cannot find any versions for core platform #{core_platform}: #{friendly_error_msg}"
       end
-
-      # get all the available distro versions
+      # Collect available distro versions
       distro_versions_available = build_map[core_platform].keys
-
-      # if core_platform is amazon then remove versions less than 2022 because they are remapped to RHEL versions
-      # and should not be eligible for consideration
-      distro_versions_available.delete_if{ |v| v.to_i < 2022 } if core_platform == "amazon"
-
-      # select only the packages from the distro versions that are <= the version we are looking for
-      # we do not want to select el7 packages for el6 platform
-      distro_versions_available.select! {|v| dsl.new_platform_version(core_platform, v, target_architecture) <= target_platform }
-
-      if distro_versions_available.length == 0
-        raise InvalidDownloadPath, "Cannot find any available #{OmnitruckDist::CLIENT_NAME} versions for this platform version #{target_platform.mapped_name} #{target_platform.mapped_version}: #{friendly_error_msg}"
+      # Remove versions that are less than 2022 if the platform is amazon
+      distro_versions_available.delete_if { |v| v.to_i < 2022 } if core_platform == "amazon"
```

Comment:
just want to double-check, this shouldn't inadvertently apply to amazon 2. in which case we can update the condition to something like `{ |v| v.to_i < 2022 && v.to_i != 2 }`

Reply:
good catch, will update this.

Reply:
So to answer this, it actually is needed because the map we have to delete it, then statically assign it in the filtered_versions var.